### PR TITLE
Small archiving refactors for GA import

### PR DESCRIPTION
**Pull request by diosmosis**
This are some small changes needed for the google analytics importer plugin. Mostly it's just making some methods public so they can be reused. Includes the new goal type PR, though that will probably change soon. #6095

### CODE REVIEW

**Comment by mattab** [comment]
Be great to review/merge this soon so we can start beta testing our [Google Analytics importer](https://matomo.org/docs/google-analytics-importer/) :rocket:

File: core/Tracker/GoalManager.php [hunk 1]
```diff
@@ -195,6 +207,42 @@ public function detectGoalMatch($goal, Action $action)
...
+    private function detectNumericGoalMatch($goal, Action $action, VisitProperties $visitProperties, Request $request)
+    {
+        switch ($goal['match_attribute']) {
+            case 'visit_duration':
+                $firstActionTime = $visitProperties->getProperty('visit_first_action_time');
+                if (empty($firstActionTime)) {
+                    return null;
+                }
+
+                $visitDurationInSecs = $request->getCurrentTimestamp() - ((int) $firstActionTime);
```
Comment (tsteur) [inline core/Tracker/GoalManager.php]: I haven't tested but wouldn't `visit_first_action_time ` be a date like `2019-02-03 04:05:06`?

File: core/Tracker/GoalManager.php [hunk 2]
```diff
@@ -195,6 +207,42 @@ public function detectGoalMatch($goal, Action $action)
...
+            default:
+                return null;
+        }
+
+        $pattern = (int) $goal['pattern'];
+
+        Common::printDebug("Matching {$goal['match_attribute']} (current value = $valueToMatchAgainst, idGoal = {$goal['idgoal']}) {$goal['pattern_type']} $pattern.");
+
+        switch ($goal['pattern_type']) {
+            case '>':
```
Comment (tsteur) [inline core/Tracker/GoalManager.php]: As mentioned in the other goal PR be good to use `greater_than` or something similar for consistency and readability.

**Comment by tsteur** [comment]
Changes are OK besides the mentioned comments

**Pull request merged**